src/timepicker/timepicker.ts

```diff
@@ -158,22 +170,28 @@ export class NgbTimepicker implements ControlValueAccessor {
   }
 
   updateHour(newVal: string | number) {
-    this.model.updateHour(toInteger(newVal) || 0);
-    this.propagateModelChange();
+    if (!this.disabled && !this.readonly) {
```

Comment:
Do we need this check? If an input is disabled / readonly a user shouldn't be able to interact with a control anyway. Or is this check here for unit-tests only? If this is just for unit tests we should change them to check if a field properly gets `readonly` / `disabled` properties.

The same comment applies to all the remaining `update`: `updateHour`, `updateMinute`, `updateSecond`